modules/src/core/ics02_client/context.rs

```diff
@@ -78,6 +80,79 @@ pub trait ClientReader {
     fn client_counter(&self) -> Result<u64, Error>;
 }
 
+pub trait ConsensusReader {
+    fn consensus_state(
+        &self,
+        client_id: &ClientId,
+        height: Height,
+    ) -> Result<Box<dyn ConsensusState>, Error>;
```

Comment:
same for ClientState, https://github.com/informalsystems/ibc-rs/blob/b11e440eea2049bf96369f5cc29d74cda06e8a86/modules/src/core/ics02_client/context.rs#L17-L19

Reply:
We've been thinking about this as well, but there are occasions where we want light-client and host implementations to be able to work with any type that implements a particular trait, and we found that the API is more convenient and simpler when we use trait objects, and so we want to avoid assoc types to keep the traits object safe.